static/js/utils.ts

```diff
@@ -264,3 +265,37 @@ export function adjustInputValue(value: number, schema: Record<string, unknown>)
 
   return value;
 }
+
+type WoTOperation = keyof typeof Constants.WoTOperation;
+type WoTFormOperation = WoTOperation | WoTOperation[] | undefined;
+/**
+ * Finds the gateway api endpoint for a specific operation.
+ * It uses the assumption that gateway enpoints are pushed to
+ * the form array as the last element.
+ *
+ * @param {Array} forms The list of forms.
+ * @param {string} operation
+ * @returns {string | undefined} the href of the select form or undefined if not found.
+ */
+export function selectFormHref(
+  forms: Array<{ href: string; op: WoTFormOperation }>,
+  operation: WoTOperation
+): string | undefined {
+  return [...forms].reverse().find((selectedForm) => {
+    try {
+      const { protocol } = new URL(selectedForm.href);
+      return (
+        protocol === 'http:' &&
+        (!selectedForm.op || selectedForm.op === operation || selectedForm.op?.includes(operation))
+      );
+    } catch (error) {
+      if (error instanceof TypeError) {
+        // URL is relative or not well formatted
+        return (
+          !selectedForm.op || selectedForm.op === operation || selectedForm.op?.includes(operation)
+        );
+      }
+      throw error;
+    }
+  })?.href;
+}
```

Comment:
Is this file the right place to introduce this utility function?

I use the spread operator and the `reverse` function, although it might be not so efficient for a large `forms` array I think it is more expressive than a `for` loop that searches the `form` backward. 

About the selection logic, tell me if you find it satisfactory. I choose to check also for protocol, maybe is an overkill? wot-adapter may publish TDs with other protocols in forms..

Reply:
I also wanted to provide unit tests for this particular function, but it is not easy to test the frontend code so I gave up. Feel free to suggest how to properly test this.